### PR TITLE
Add Knative GSoC ideas

### PR DESCRIPTION
**Pull request by aliok**
Add Knative GSoC ideas we collected in our ideas page here: https://github.com/knative/community/blob/main/google-summer-of-code/gsoc-2023.md

### CODE REVIEW

File: summerofcode/2023.md
Comment (aliok) [inline summerofcode/2023.md:33]: I've added this TOC as well. Let me know if you would like me to remove it.

**Review by caniszczyk — APPROVED** [review]

File: summerofcode/2023.md
Comment (nate-double-u) [reply summerofcode/2023.md:33]: I appreciate the TOC. Usually i add it when we move projects from ideas to accepted. If folks update it as they add ideas i'm happy to leave it in, but if folks don't then I'd rather take it out and add it all at once with a tool at the end.

File: summerofcode/2023.md
Comment (nate-double-u) [inline summerofcode/2023.md:72]: We need to add a maintainer to this project in addition to these mentors -- forgive me if one of these three are already, we may need to update the lists.

Comment (nate-double-u) [reply summerofcode/2023.md:72]: We also need mentor email addresses to add folks to the system when the time comes. Could you add them @aliok?

Comment (aliok) [reply summerofcode/2023.md:72]: > We need to add a maintainer to this third project in addition to these mentors -- forgive me if one of these three are already, we may need to update the lists.

@nate-double-u They are all maintainers. Is there a system or something like that at CNCF side where they are not listed? Where are those lists, if it is a public list?

Comment (aliok) [reply summerofcode/2023.md:72]: > We also need mentor email addresses to add folks to the system when the time comes. Could you add them @aliok?

@nate-double-u sorry about that. I was checking the GSOC 22 doc and didn't see any email addresses in that doc.

Would you like to see the email addresses in the doc? ...which might be bad because of spam... or in the list you mentioned previously? I can provide them to you in Slack DM.

Comment (nate-double-u) [reply summerofcode/2023.md:72]: > > We need to add a maintainer to this third project in addition to these mentors -- forgive me if one of these three are already, we may need to update the lists.
> 
> @nate-double-u They are all maintainers. Is there a system or something like that at CNCF side where they are not listed? Where are those lists, if it is a public list?

This is the list I use to check against: https://github.com/cncf/foundation/blob/main/project-maintainers.csv

Comment (nate-double-u) [reply summerofcode/2023.md:72]: > > We also need mentor email addresses to add folks to the system when the time comes. Could you add them @aliok?
> 
> @nate-double-u sorry about that. I was checking the GSOC 22 doc and didn't see any email addresses in that doc.
> 
> Would you like to see the email addresses in the doc? ...which might be bad because of spam... or in the list you mentioned previously? I can provide them to you in Slack DM.

I'd prefer all the data in one place, but I'm sympathetic to privacy concerns. If there are public email addresses please share them here, otherwise please send them via DM.

Comment (evankanderson) [reply summerofcode/2023.md:72]: One thing which may be tripping up the system is that Knative emulated both Kubernetes and Istio's governance model with a TOC/Steering Committee and smaller sub-area leads (called working group leads in Knative).  We also have individuals who are approvers for code but aren't in a leadership role.  The working group leads are similar to SIG leads in Kubernetes, but I'm not sure if they or approvers for specific areas of the project are in that maintainers file.

Comment (aliok) [reply summerofcode/2023.md:72]: @nate-double-u  @evankanderson 

I am a bit confused about this CSV file and found other people got similar confusions :) I found this ticket and added a comment there: https://github.com/cncf/foundation/issues/433#issuecomment-1410006312

@nate-double-u The issue is, mentors listed in that idea are not in Knative's steering committee or in technical oversight committee. Hence, they're not in that CSV file. However, they're the "approvers" for the component for the project idea.

Similar situation applies to me, with the first idea ("Multiple Knative Eventing control planes").

As I wrote in my comment (https://github.com/cncf/foundation/issues/433#issuecomment-1410006312), it might be good to specify who can be listed as the mentors. Or, if mentoring WG needs sponsorship/approval from the people in that CSV file.

Comment (aliok) [reply summerofcode/2023.md:72]: Having said these about improving the process here, switching back to practical side... Mentors listed in the first and the third idea are the approvers (SMEs) for the relevant components and they can provide mentorship in GSoC.

**Pull request merged**
